### ISSUE TITLE
[Bug]: Control border not fully visible with headless false (Firefox)

### ISSUE THREAD

**Issue by kacperczerwinski** [microsoft/playwright#31869]
### Version

1.45.3

### Steps to reproduce

We ran into a problem with my team with regression tests in Firefox. Firefox hides scrollbars in headless mode. If I set headless to false in the configuration, the scrollbar is visible but the images are not stable. Is there a way to override the css to show the scrollbars in headless mode or even better an option like ignoreDefaultArgs: ['--hide-scrollbars'] for chrome?

1. Clone my repo https://github.com/kacperczerwinski/example-playwright
2. You will find generated example screenshots under **tests/example.spec.ts-snapshots.** You can check the example-with-headless-false-firefox-win32 screenshot where the control doesn't look good but has scrollbar
3. You can check example-with-headless-true-firefox-win32 control looks good but missing scrollbar
4. You can check example-with-hide-scrollbars-ignored-chromium-win32 which is how it should look like in Firefox (scrollbar and not missing card lines)
5. In summary. What we want to achieve is that the images look good in Firefox (nothing is missing and with a scrollbar) as in chromium

### Expected behavior

Having screenshots with fully visible controls and scrollbars in firefox

### Actual behavior

It's only possible to have scrollbars in headless set to false but that makes the control missing parts of the border

### Environment

```shell
System:
    OS: Windows 11 10.0.22621
    CPU: (16) x64 12th Gen Intel(R) Core(TM) i7-1270P
    Memory: 14.65 GB / 31.24 GB
  Binaries:
    Node: 18.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 9.5.1 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.15.6 - C:\Program Files\nodejs\pnpm.CMD
  IDEs:
    VSCode: 1.91.1 - C:\Program Files\Microsoft VS Code\bin\code.CMD
  Languages:
    Bash: 5.1.16
  npmPackages:
    @playwright/test: ^1.45.3 => 1.45.3 
    playwright: ^1.45.3 => 1.45.3
```

**Comment by pavelfeldman** [comment]
Not sure I follow. We hide scroll bars because it makes screenshots less stable. You enable them manually, but it makes your screenshots less stable. Please provide more details, screenshots, exact actual and expectations.

**Comment by kacperczerwinski** [comment]
Thanks for your response. The scrollbars doesn't make our screenshots less stable but making them in headless mode set to false. Like here, the image on left is made with headless false and you can realize that there is a missing line in the text area bottom bar, with headless true looks fine
<img width="679" alt="Untitled" src="https://github.com/user-attachments/assets/54e40ff6-134f-4373-bcdd-60141efde82a">


Basically what we want is having screenshots in headless mode set true (is default I think) with scrollsbars in firefox something like  like launchOptions: { ignoreDefaultArgs: ['--hide-scrollbars'] } for firefox.

**Comment by mxschmitt** [comment]
I'm taking over the triaging for this week. Is hard to follow for me this issue, so I recommend to follow our bug template with providing a reproduction which we can run locally, expected, and actual outcome.

**Comment by kacperczerwinski** [comment]
Thank you for you response @mxschmitt! I just updated the description. Added steps to reproduce the problem. I hope now it's clear

**Comment by mxschmitt** [comment]
Awesome, thank you. Its expected as of today, that `ignoreDefaultArgs: ['--hide-scrollbars']` is not working with Firefox.

We already have a issue for having scrollbars in Firefox headless. Let me fold it into https://github.com/microsoft/playwright/issues/4295.
